File: src/main/java/com/mailbox/BoxTester.java
package com.mailbox;

import java.util.ArrayList;
import java.util.Scanner;

public class BoxTester {
    public static void main(String[] args) {
        boolean hasMatchBoxes = false;
        ArrayList<MailBox> boxes = initAllTypeOfBoxes();
        Scanner scanner = new Scanner(System.in);

        System.out.println("Please enter object's length");
        float length = scanner.nextFloat();
        System.out.println("Please enter object's width");
        float width = scanner.nextFloat();
        System.out.println("Please enter object's height");
        float height = scanner.nextFloat();

        for (MailBox box : boxes) {
            if (box.validate(length, width, height)) {
                box.printType();
                hasMatchBoxes = true;
            }
        }

        if (!hasMatchBoxes) {
            System.out.println("Sorry, we don't have any proper box for you!");
        }
    }

    private static ArrayList<MailBox> initAllTypeOfBoxes() {
        ArrayList<MailBox> boxes = new ArrayList<>();
        MailBox box3 = new Box3();
        MailBox box5 = new Box5();
        boxes.add(box3);
        boxes.add(box5);
        return boxes;
    }
}

File: src/main/kotlin/com/mailbox/kt/BoxTester.kt
package com.mailbox.kt

import java.util.*
import kotlin.collections.ArrayList

fun main() {
    var hasMatchBox = false
    val boxes = initAllTypeOfBoxes()
    val scanner = Scanner(System.`in`)

    println(" Please enter object's length: ")
    val length = scanner.nextFloat()
    println(" Please enter object's width: ")
    val width = scanner.nextFloat()
    println(" Please enter object's height: ")
    val height = scanner.nextFloat()

    boxes.forEach { box: MailBox ->
        if (box.validate(length, width, height)) {
            box.printType()
            hasMatchBox = true
        }
    }

    if (!hasMatchBox) {
        println("Sorry, we don't have any proper box for you!")
    }
}

fun initAllTypeOfBoxes(): ArrayList<MailBox> {
    val boxes = arrayListOf<MailBox>()
    val box3 = Box3()
    val box5 = Box5()
    boxes.add(box3)
    boxes.add(box5)
    return boxes
}